nexus/src/sagas.rs

```diff
@@ -475,9 +488,77 @@ async fn sic_create_network_interfaces_undo(
     Ok(())
 }
 
+async fn sic_attach_disks_to_instance(
+    sagactx: ActionContext<SagaInstanceCreate>,
+) -> Result<(), ActionError> {
+    ensure_instance_disk_attach_state(sagactx, true).await
+}
+
+async fn sic_attach_disks_to_instance_undo(
+    sagactx: ActionContext<SagaInstanceCreate>,
+) -> Result<(), anyhow::Error> {
+    Ok(ensure_instance_disk_attach_state(sagactx, false).await?)
+}
+
+async fn ensure_instance_disk_attach_state(
+    sagactx: ActionContext<SagaInstanceCreate>,
+    attached: bool,
+) -> Result<(), ActionError> {
+    let osagactx = sagactx.user_data();
+    let saga_params = sagactx.saga_params();
+    let opctx =
+        OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
+
+    let saga_disks = &saga_params.create_params.disks;
+    let instance_name = sagactx.lookup::<db::model::Name>("instance_name")?;
+
+    let organization_name: db::model::Name =
+        saga_params.organization_name.clone().into();
+    let project_name: db::model::Name = saga_params.project_name.clone().into();
+
+    for disk in saga_disks {
+        match disk {
+            params::InstanceDiskAttachment::Create(_) => {
+                todo!();
```

Comment:
(You don't really need to act on this, but) I'm pretty sure that trying to create a disk at the same time as an instance is going to require more steps in the saga - we'd likely need to make the instance creation saga a bit longer, and perform the creation actions as "unconditional nodes in the saga graph, containing conditionally-executed logic".